src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -158,17 +159,19 @@ private int getLowestEdgePriority(List<T> nodes) {
     return lowerPriority == null ? 0 : lowerPriority;
   }
 
-  private int getLowestOrder(List<T> nodes) {
+  private int getLowestOrder(List<T> nodes, int weight) {
```

Comment:
`order` is the way to have a hierarchy between edges with the same weight. It was introduced by @krmahadevan with https://github.com/cbeust/testng/commit/55c3745502fd9d9e23e49f0bdafc577cacb1fc20
and has only sense with its associated weight.
If someone has better names, I agree to change them.